Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/EventLogServiceImpl.java

```diff
@@ -565,4 +573,30 @@ public boolean saveFileToVolume(String filename, String prefix, ByteArrayOutputS
 	return true;
     }
 
+	@Override
+	public List<String> getPerspectiveTagByLog(Integer logId) {
+
+		List<String> perspectiveList = new ArrayList<>();
+
+		Set<Usermetadata> usermetadataSet = userMetadataService.getUserMetadataByLog(logId,
+				UserMetadataTypeEnum.PERSPECTIVE_TAG);
+
+		if (usermetadataSet.isEmpty()) {
+			LOGGER.info("Log (ID: {}) doesn't have associated perspectives stored in DB.", logId);
+			return perspectiveList;
+		}
+
+		String jsonString = usermetadataSet.iterator().next().getContent();
+		ObjectMapper objectMapper = new ObjectMapper();
+
+		try {
+			perspectiveList = objectMapper.readValue(jsonString, new TypeReference<>() {
+			});
+		} catch (JsonProcessingException e) {
+			LOGGER.error("Could not deserialize JSON content from given JSON content String: " + jsonString, e);
+		}
```

Comment:
This exception should be handled properly to indicate something wrong in the data for timely repair. As commented above, suggest to throw a domain-specific exception instead of an implementation-specific one: UsermetadataException. The UI layer will need to catch and display the error message properly, e.g. "Internal error when processing log metadata for perspectives".